services/frontend-v3/src/components/profileForms/primaryInfoForm/gedsUpdateModal/GedsUpdateModalView.jsx

```diff
@@ -0,0 +1,319 @@
+    savedValue: savedProfile.telephone ? savedProfile.telephone : "-",
+    gedsLabel: gedsProfile.telephone ? gedsProfile.telephone : "-",
+    gedsValue: gedsProfile.telephone ? gedsProfile.telephone : "-",
+    paramName: "telephone",
+  },
+  {
+    key: "4",
+    rowName: <FormattedMessage id="profile.location" />,
+    savedLabel: savedProfile.officeLocation
+      ? `${savedProfile.officeLocation.streetNumber} ${savedProfile.officeLocation.streetName}, ${savedProfile.officeLocation.city}`
+      : "-",
+    savedValue: savedProfile.officeLocation
+      ? savedProfile.officeLocation.id
+      : "-",
+    gedsLabel: gedsProfile.locationName ? gedsProfile.locationName : "-",
+    gedsValue: gedsProfile.locationId ? gedsProfile.locationId : "-",
+    paramName: "location",
+  },
+  {
+    key: "5",
+    rowName: <FormattedMessage id="profile.career.header.name" />,
+    savedLabel: savedProfile.jobTitle ? savedProfile.jobTitle : "-",
+    savedValue: savedProfile.jobTitle ? savedProfile.jobTitle : "-",
+    gedsLabel: gedsProfile.jobTitle ? gedsProfile.jobTitle[locale] : "-",
+    gedsValue: gedsProfile.jobTitle ? gedsProfile.jobTitle[locale] : "-",
+    paramName: "jobTitle",
+  },
+  {
+    key: "6",
+    rowName: <FormattedMessage id="profile.branch" />,
+    savedLabel: savedProfile.branch,
+    savedValue: savedProfile.branch,
+    gedsLabel: gedsProfile ? gedsProfile.branch[locale] : "-",
+    gedsValue: gedsProfile ? gedsProfile.branch[locale] : "-",
+    paramName: "branch",
+  },
+  {
+    key: "7",
+    rowName: <FormattedMessage id="profile.org.tree" />,
+    savedLabel: savedProfile.organizations[0]
+      ? savedProfile.organizations[0][savedProfile.organizations[0].length - 1]
+          .title
+      : "-",
+    savedValue: savedProfile.organizations[0]
+      ? savedProfile.organizations[0][savedProfile.organizations[0].length - 1]
+          .title
+      : "-",
+    gedsLabel: gedsProfile
+      ? gedsProfile.organizations[0][gedsProfile.organizations[0].length - 1]
+          .title[locale]
+      : "-",
+    gedsValue: gedsProfile
+      ? gedsProfile.organizations[0][gedsProfile.organizations[0].length - 1]
+          .title[locale]
+      : "-",
+    paramName: "organization",
+  },
+];
+
+/**
+ * Component to render the GC directory Sync modal
+ * @param {Boolean} visibility - visibility of modal.
+ */
+const GedsUpdateModalView = ({ visibility }) => {
+  const axios = useAxios();
+  const [newGedsValues, setNewGedsValues] = useState(null);
+  const [tableData, setTableData] = useState(null);
+  const [tableLoading, setTableLoading] = useState(false);
+  const [errorCaught, setErrorCaught] = useState(false);
+  const { locale } = useSelector((state) => state.settings);
+  const { id, name } = useSelector((state) => state.user);
+
+  /**
+   * Save Data to DB by sending to backend API
+   * @param {Object} formValues - Data from primary info form.
+   */
+  const saveDataToDB = async (formValues) => {
```

Comment:
I would maybe move this to the controller

Reply:
done